python_modules/dagster/dagster/_core/execution/context/asset_execution_context.py

```diff
@@ -458,16 +458,42 @@ def add_output_metadata(
         output_name: Optional[str] = None,
         mapping_key: Optional[str] = None,
     ) -> None:
-        return self.op_execution_context.add_output_metadata(
-            metadata=metadata, output_name=output_name, mapping_key=mapping_key
+        return self._step_execution_context.add_output_metadata(
+            metadata=metadata,
+            output_name=output_name,
+            mapping_key=mapping_key,
+        )
+
+    @public
+    def add_asset_metadata(
+        self,
+        metadata: Mapping[str, Any],
+        asset_key: Optional[CoercibleToAssetKey] = None,
+        partition_key: Optional[str] = None,
+    ) -> None:
+        """Add metadata to an asset materialization event. This metadata will be available in the
+        Dagster UI and can be used to filter and search for assets.
```

Comment:
It can't _really_ be used to filter and search for assets (except in dagster-plus, kinda). Maybe just ditch that part